meta/engineering_principles.md

```diff
@@ -0,0 +1,146 @@
+# Engineering Principles
+
+We have [company values](what_is_artsy.md), and are working on our shared values with design and product. However,
+when we make decisions on what and how to do engineering, we need guiding principles. Things that make Artsy's
+development culture unique and drive us to do great work.
+
+### Open Source by Default
+
+We consider that it is the data and relationships with the art world what makes Artsy a difficult business to copy.
+Our code only represents ideals of our past and close-to present. A code project should start as open, then move to
+closed if there are valid business needs that require it to be private.
+
+Working in the open allows individual to retain their work and ideas post-Artsy. We can share the exact problems we
+see with others, and provide great high ways to teach via direct references.
+
+Expectations for our OSS _apps_ are different than OSS _libraries_. An app has little-to-no desire for community
+contributions. Instead, it's open as a reference. With tools and libraries we can build communities around, but
+most of our projects don’t get large enough for their stewardship to become burdensome.
+
+- Examples:
+
+  - artsy/meta
+  - artsy/force
+  - artsy/eigen
+  - artsy/emission
+
+- Further Reading:
+
+  - https://artsy.github.io/series/open-source-by-default/
+  - https://ashfurrow.com/blog/open-source-ideology/
+  - https://code.dblock.org/2015/02/09/becoming-open-source-by-default.html
+  - https://www.objc.io/issues/22-scale/artsy/
+
+### Own Your Dependencies
```

Comment:
I see two different but related concepts here: "own your dependencies" and "don't reinvent the wheel." You could in theory have one without the other but they go together pretty naturally. Should they be separate, or maybe described with a more general name?

Reply:
I think own your dependencies at core isn’t quite the same as don’t reinvent the wheel - this speaks to the acceptance that all the code in your stack is something you should contribute to

Don’t reinvent the wheel is a pretty good  way of thinking but it’s basically something most engineering teams should do - not sure if it is something unique enough to us to warrant inclusion

Reply:
Other than a sentence nod to it, probably

Reply:
Yeah, I see "don't reinvent the wheel" as the foundation that the "own your own dependencies" is built upon. A nod sounds like a good idea.

Reply:
Agreed that "don't reinvent the wheel" should be baseline but in practice lots of teams end up reinventing the wheel because it's personally more gratifying, or because they believe their needs are unique enough to warrant it. That we take it seriously is kind of interesting!

Reply:
Yeah! In contrast to the "not invented here" idea, the objc.io article linked to describes the mobile team's ideology of "proudly discovered elsewhere", which I still admire.

Reply:
We should re-use that phrase here